sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/spi/HttpClientProvider.java

```diff
@@ -0,0 +1,8 @@
+package com.azure.core.implementation.http.spi;
+
+import com.azure.core.http.HttpClient;
+
+public interface HttpClientProvider {
+
```

Comment:
nit: empty new line.

Also preference for `createInstance()`

Reply:
Maybe it is just me, but I always leave the first line of a class empty if there are no fields in the class..it feels better to 'leave space' for the fields, and makes it easier to read the code.